nifi-mock/src/main/java/org/apache/nifi/util/TestRunner.java

```diff
@@ -865,4 +866,21 @@
      * @return the State Manager that is used to store and retrieve state for the given controller service
      */
     MockStateManager getStateManager(ControllerService service);
+
+    /**
+     *  Maybe we should add possibility to be more precise:
+     *      "All FlowFiles must meet all conditions"
+     *  or
+     *      "At least one FlowFile must meet all conditions"
+     *  or
+     *      "Each FlowFile should meet at least one condition"
+     *
+     *  Current functionality is: "Each FlowFile should meet at least one condition"
+     *  So instead of assertAllConditionsMet we should use something like assertFlowFileMeetAnyCondition
+     *  Or add extra parameter which specifies how FlowFile must meet conditions.
+     *
+     */
+    void assertAllConditionsMet(final String relationshipName, ConditionsBuilder... andContentEqual);
```

Comment:
Predicates is a good idea.

“a very valuable method/test would be to indicate that all conditions are met exactly once.”
You mean exactly one (only one) FlowFile meets all conditions.
runner.assertExactlyOnceAllConditionsMet(MY_RELATIONSHIP, ...